hackster-service-client/src/test/java/com/lohika/jclub/hackster/client/HacksterServiceClientTest.java

```diff
@@ -0,0 +1,65 @@
+package com.lohika.jclub.hackster.client;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.LogMessageWaitStrategy;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = HacksterServiceClientTestApplication.class)
+@ContextConfiguration(initializers = HacksterServiceClientTest.Initializer.class)
+public class HacksterServiceClientTest {
+
+  @ClassRule
+  public static GenericContainer HacksterService = new GenericContainer("hackster-service:latest")
+      .withExposedPorts(8082)
+      .withEnv("maxAllowedApartmentsPerRealtor", "4")
+      .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Started HacksterServiceApplication in.*\\s"));
+
+  @Autowired
+  private HacksterServiceClient hacksterServiceClient;
+
+  @Test
+  public void checkIfNewNumberIsNotHackster() {
+    boolean isHackster = hacksterServiceClient.isHackster("123123123");
+
+    assertThat(isHackster, equalTo(false));
+  }
+
+  @Test
+  public void checkIfOldNumberIsNotHackster() {
+    boolean isHacksterFalse = hacksterServiceClient.isHackster("321321312");
+
+    for (int i = 0; i < 5; i++) {
+      hacksterServiceClient.isHackster("321321312");
+    }
+
+    boolean isHacksterTrue = hacksterServiceClient.isHackster("321321312");
+
+    assertThat(isHacksterFalse, equalTo(false));
```

Comment:
Because it's clarify that number which wasn't hackster on beginning become a hackster after several calls